cordova-plugin-one/src/android/OnePlugin.java

```diff
@@ -8,419 +8,451 @@
-                    }
-                    try {
-                        one.sendPushToken(stringURL);
-                        callbackContext.success();
-                    } catch (Exception e) {
-                        callbackContext.error("Failed to handle URL");
-                    }
-                }       
-            });
-            return true;
-        } else if ("getPushToken".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    try {
-                        String token = one.getPushToken();
-                        callbackContext.success(token);
-                    } catch (Exception e) {
-                        callbackContext.error("Failed to get push token");
-                    }
-                }
-            });
-            return true;
-        } else if ("enablePushNotifications".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    boolean enable = args.optBoolean(0);
-                    one.enablePushNotifications(enable);
-                    callbackContext.success();
-                }
-            });
-            return true;  
-        } else if ("whitelistIdentityTransferLinks".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    ArrayList<String> links = getStringListFromCordovaArgs(args);
-                    if (links.size() > 0) {
-                        one.whitelistIdentityTransferLinks(links);
-                        callbackContext.success();
-                    } else {
-                        callbackContext.error("The SDK cannot whitelist any links as no links were provided.");
-                    }
-                }
-            });
-            return true;
-        } else if ("blacklistIdentityTransferLinks".equals(action)) {
-            cordova.getActivity().runOnUiThread(new Runnable() {
-                public void run() {
-                    ArrayList<String> links = getStringListFromCordovaArgs(args);
-                    if (links.size() > 0) {
-                        one.blacklistIdentityTransferLinks(links);
-                        callbackContext.success();
-                    } else {
-                        callbackContext.error("The SDK cannot blacklist any links as no links were provided.");
-                    }
-                }
-            });
-            return true;
+  protected static final String LOG_TAG = "OnePlugin";
+
+  @Override
+  public boolean execute(final String action, final CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
+    if ("initializeOne".equals(action)) {
+      return initializeOne(args, callbackContext);
+    } else if ("sendInteraction".equals(action)) {
+      cordova.getActivity().runOnUiThread(() -> {
```

Comment:
do we need to `runOnUiThread` for these to work as expected?

Reply:
No we do not.  Removed